larkspur/larkspur.py

```diff
@@ -389,23 +392,26 @@ def __contains__(self, key: Any) -> bool:
                 return True
         return False
 
-    def add(self, key: Any):
-        """Checks the current count of the bloom filter against the capacity, then scales up when needed.
-        Adds the item key into the bloom filter.
+    def add(self, key: KeyType) -> bool:
+        """Adds the item key into the bloom filter.
 
         Args:
-            key: a item need to be added.
+            key: The item to be added.
 
         Returns:
-            A boolean returns true indicating all bits are already set to 1, 
-            otherwise sets the bits to 1 and increase count by 1.
-            Or an IndexError when current count reaches the capacity.
+            True if key exists and False if it not and it was added
         """
-        # Race conditions may occur
+        # Check to see if any filters already contain the key.
+        # This check is necessary because self._get_next_filter will return the latest 
+        # BloomFilter which may not contain this key, but the key may still exist
+        # in an earlier BloomFilter if initial_capacity was exceeded.
+        if self.__contains__(key):
```

Comment:
you can use the `in` syntax since the __contains__ method is present... I think

if `key in self`